### DISCUSSION TITLE
MultiSelectDropDown - add role property

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#242] [Ideas]
As with the native Anvil components, sometimes it is useful to be able to add a role to the MultiSelectDropDown - ie a 'mandatory' role to outline it in red or similar.

**Comment by s-cork** [comment]
@rhurlbatt did you want to implement this?

**Reply by rhurlbatt** [reply]
Yes I would like to attempt this

**Reply by s-cork** [reply]
Awesome. First is to decide if you want the role on just the main component or also on the dropdown list too. 

I think just the main component is a good idea. 
The drop down list is not part of the component hierarchy so that'll get into a whole load of Js you probably doesn't want to think about. 

There are already components that support a role property in anvil- extras. You can find one and use that code. I think tabs might. 

Then see if you can make the role do what you want on an actual example.

**Reply by rhurlbatt** [reply]
Cool - thank you for the pointers. I had already found the Tabs components role so was planning on replicating some of this.

I also think the main component would do - and no I don't want to look at any JS 🤢 

I will have a look at this today
